.github/workflows/delete-pr-build-on-close.yml

```diff
@@ -19,17 +19,21 @@ jobs:
         run: |
           # Figure out tag name from branch name. This is coupled to the tag name generation that exists in .circleci/config.yml's `create-github-release` job.
           RELEASES=$(gh release list --repo="slackapi/slack-cli" --order="desc" --json="tagName" --exclude-drafts --exclude-pre-releases --limit=24 --jq ".[] | .tagName")
+          RELEASE_FOUND=1
           for TAGS in $RELEASES; do
             TAG_NAME="${TAGS}-${{ github.event.ref }}"
             echo "Identified pre-release tagname to 🔪: $TAG_NAME"
 
             # Delete the pre-release
             if GH_DEBUG=1 gh release --repo="slackapi/slack-cli" delete "$TAG_NAME" -y --cleanup-tag; then
               echo "Successfully deleted $TAG_NAME"
-              exit 0
+              RELEASE_FOUND=0
             else
               echo "Failed to find $TAG_NAME, trying next..."
             fi
+            sleep 1
```

Comment:
😴 Rest is so important and I hope this helps us avoid rate limits of rapid loops!
https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#about-secondary-rate-limits